bentoml/cli/bento_service.py

```diff
@@ -96,20 +88,33 @@ def run_with_conda_env(bundle_path, command):
     return
 
 
-def make_bento_name_docker_compatible(name, tag):
+def make_bento_name_docker_compatible(name, version):
     """
     Name components may contain lowercase letters, digits and separators.
     A separator is defined as a period, one or two underscores, or one or more dashes.
 
-    A tag name must be valid ASCII and may contain lowercase and uppercase letters,
-    digits, underscores, periods and dashes. A tag name may not start with a period
+    The version must be valid ASCII and may contain lowercase and uppercase letters,
+    digits, underscores, periods and dashes. Version may not start with a period
     or a dash and may contain a maximum of 128 characters.
 
     https://docs.docker.com/engine/reference/commandline/tag/#extended-description
     """
     name = name.lower().strip("._-")
-    tag = tag.lstrip(".-")[:128]
-    return name, tag
+    version = version.lstrip(".-")[:128]
+    return name, version
+
+
+def validate_tag(ctx, param, tag):  # pylint: disable=unused-argument
```

Comment:
I think the ideal behavior of `validate_tag` is telling the user that they are using a wrong format for docker tag instead of converting it silently. This matches `docker build` cli's behavior as well:

![Screen Shot 2020-07-11 at 3 11 46 PM](https://user-images.githubusercontent.com/489344/87234730-d7b2a880-c388-11ea-94cf-1fb32a271957.png)